test/integration/fake_upstream.h

```diff
@@ -324,10 +379,13 @@ class FakeHttpConnection : public Http::ServerConnectionCallbacks, public FakeCo
 
   FakeHttpConnection(SharedConnectionWrapper& shared_connection, Stats::Store& store, Type type);
   // By default waitForNewStream assumes the next event is a new stream and
-  // fails an assert if an unexpected event occurs. If a caller truly wishes to
-  // wait for a new stream, set ignore_spurious_events = true.
-  FakeStreamPtr waitForNewStream(Event::Dispatcher& client_dispatcher,
-                                 bool ignore_spurious_events = false);
+  // returns AssertionFaliure if an unexpected event occurs. If a caller truly
+  // wishes to wait for a new stream, set ignore_spurious_events = true.
+  ABSL_MUST_USE_RESULT
+  testing::AssertionResult
+  waitForNewStream(Event::Dispatcher& client_dispatcher, FakeStreamPtr* stream,
```

Comment:
Our style guide prefers using a reference for connection here, since it can't be nullptr. Also add a comment indicating that the connection is returned via the reference. (Likewise for waitForRawConnection and waitForHttpConnection.)